pkg/lookable/asg.go

```diff
@@ -56,11 +56,13 @@ func (asg AutoScalingGroup) LookupIPs(ipv6 bool) ([]string, error) {
 		return output, nil
 	}
 
-	// Make a list of instance ID
-	instances := make([]*string, numInstances)
-	for i, inst := range resp2.AutoScalingGroups[0].Instances {
-		instances[i] = inst.InstanceId
-	}
+        // Make a list of healthy instance ID in the ASG
+        instances := make([]*string, numInstances)
+        for i, inst := range resp2.AutoScalingGroups[0].Instances {
+                if (*inst.HealthStatus == "Healthy" && *inst.LifecycleState == "InService") {
+                        instances[i] = inst.InstanceId
```

Comment:
You created the `instances` slice with an initial length. Some elements will not be assigned and left with the default value (empty string). No issue when calling the `DescribeInstances` function with it?

You could have created an empty slide and append instances to it when they are healthy.

Reply:
should be better now